packages/init/src/index.ts

```diff
@@ -11,24 +11,32 @@ import {
   detectFramework,
   installLib,
 } from "./steps";
-import { findArgument, logs } from "./utils";
+import { findArgument, logs, getApplicationMode, Client } from "./utils";
 
 async function init() {
   const cwd = findArgument(process.argv, "cwd") || process.cwd();
-  const base = findArgument(process.argv, "base") || CONSTS.DEFAULT_BASE;
+  const mode: ApplicationMode =
+    getApplicationMode(findArgument(process.argv, "mode")) ||
+    ApplicationMode.PROD;
   const lib: string | undefined = findArgument(process.argv, "library");
   const branch: string | undefined = findArgument(process.argv, "branch");
-  const isTrackingAvailable =
+  const isTrackingAvailable: boolean =
     findArgument(process.argv, "tracking") !== "false";
-  const maybeRepositorySubdomain = findArgument(process.argv, "repository");
+  const preSelectedRepository: string | undefined = findArgument(
+    process.argv,
+    "repository"
+  );
 
   Tracker.get().initialize(
     process.env.PUBLIC_SM_INIT_SEGMENT_KEY ||
       "JfTfmHaATChc4xueS7RcCBsixI71dJIJ",
     isTrackingAvailable
   );
 
-  void Tracker.get().trackInitStart(maybeRepositorySubdomain);
+  void Tracker.get().trackInitStart(preSelectedRepository);
+
+  // initializing the client with what we have for now.
+  Client.initialize(mode, Prismic.PrismicSharedConfigManager.getAuth());
```

Comment:
should pass this down to the other functions

Reply:
Answered above, we can still discuss it but i'm resolving this one to not have multiple comment about the same thing.